middleware/proxy/upstream.go

```diff
@@ -175,6 +181,12 @@ func parseBlock(c *parse.Dispenser, u *staticUpstream) error {
 			return c.ArgErr()
 		}
 		u.IgnoredSubPaths = ignoredPaths
+	case "proxy_redirect":
```

Comment:
What if we just call it "redirect" instead of proxy_redirect?

Also, how is this redirect different from the `redir` directive? Do you have an example of how this is used?